CovertInfiltration/Src/CovertInfiltration/Classes/X2DownloadableContentInfo_CovertInfiltration.uc

```diff
@@ -28,9 +28,57 @@ static event UpdateDLC ()
 {
 	class'XComGameState_ActivityChainSpawner'.static.Update();
 	class'XComGameState_CovertActionExpirationManager'.static.Update();
+	UpdateRemoveCovertActions();
 	UpdateShowTutorial();
 }
 
+static function UpdateRemoveCovertActions ()
+{
+	local XComGameState NewGameState;
+	local XComGameState_CovertInfiltrationInfo CIInfo;
+	local XComGameState_CovertAction ActionState;
+	local StateObjectReference ActionRef;
+	local bool bDirty;
+	
+	bDirty = false;
+	CIInfo = class'XComGameState_CovertInfiltrationInfo'.static.GetInfo();
+
+	if (CIInfo == none) return;
+
+	NewGameState = class'XComGameStateContext_ChangeContainer'.static.CreateChangeState("CI: Removing Flagged Covert Actions");
+	CIInfo = XComGameState_CovertInfiltrationInfo(NewGameState.ModifyStateObject(class'XComGameState_CovertInfiltrationInfo', CIInfo.ObjectID));
+
+	foreach CIInfo.CovertActionsToRemove(ActionRef)
+	{
+		ActionState = XComGameState_CovertAction(NewGameState.ModifyStateObject(class'XComGameState_CovertAction', ActionRef.ObjectID));
+		if (ActionState != none && ActionState.bCompleted == false)
```

Comment:
If this check fails, this piece of code would be running each tick and then getting discarded (`CovertActionsToRemove` will never be emptied)

Reply:
I don't think so, there's no `return` between that check and the `.Length = 0;` call. It goes once through the array then empties the array regardless of what's inside it and whether the contents passed the check or not.

Reply:
Yes, but if the check doesn't pass then `bDirty` remains `false`, so the `CIInfo.CovertActionsToRemove.Length = 0;` change gets nullified by the `CleanupPendingGameState` call